autoscheduler/frontend/src/tests/ui/Schedule.test.tsx

```diff
@@ -0,0 +1,107 @@
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import * as React from 'react';
+
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+import Meeting, { MeetingType } from '../../types/Meeting';
+import Section from '../../types/Section';
+import Instructor from '../../types/Instructor';
+import Schedule from '../../components/SchedulingPage/Schedule/Schedule';
+import autoSchedulerReducer from '../../redux/reducer';
+import { testSchedule3 } from '../testSchedules';
+
+const testSection = new Section({
+  id: 123456,
+  crn: 123456,
+  subject: 'CSCE',
+  courseNum: '121',
+  sectionNum: '200',
+  minCredits: 3,
+  maxCredits: null,
+  currentEnrollment: 0,
+  instructor: new Instructor({
+    name: 'Aakash Tyagi',
+  }),
+});
+
+const testMeeting1 = new Meeting({
+  id: 12345,
+  building: 'HRBB',
+  meetingDays: new Array(7).fill(true),
+  startTimeHours: 10,
+  startTimeMinutes: 20,
+  endTimeHours: 11,
+  endTimeMinutes: 10,
+  meetingType: MeetingType.LEC,
+  section: testSection,
+});
+
+describe('Schedule UI', () => {
+  describe('renders without errors', () => {
+    test('when given an empty schedule', () => {
+      // arrange and act
+      const store = createStore(autoSchedulerReducer);
+      const { container } = render(<Provider store={store}><Schedule /></Provider>);
+
+      // assert
+      expect(container).toBeTruthy();
+    });
+
+    test('when given a schedule with 1 meeting', () => {
+      // arrange and act
+      const store = createStore(autoSchedulerReducer, {
+        schedules: [[testMeeting1]], selectedSchedule: 0,
+      });
+      const { container } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+
+      // assert
+      expect(container).toBeTruthy();
+    });
+  });
+
+  describe('uses unique colors', () => {
+    test('for up to 10 different sections', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, {
+        schedules: [testSchedule3], selectedSchedule: 0,
+      });
+      const { getAllByText } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+      // as a pre-condition for this test to make sense, the test schedule cannot have
+      // more than 10 unique sections
+      const uniqueSections = new Set(testSchedule3.map((mtg) => mtg.section.id));
+      expect(uniqueSections.size).toBeLessThanOrEqual(10);
+
+      // act
+      const cardEls = getAllByText(/\w{4} \d{3}-\d{3}/);
+      const meetingCards = cardEls.map((card) => {
+        // find the colored element that is a parent of the div with text
+        let coloredEl = card;
+        while (!coloredEl.style.backgroundColor) coloredEl = coloredEl.parentElement;
+
+        return {
+          backgroundColor: coloredEl.style.backgroundColor,
+          textContent: card.textContent,
+        };
+      });
```

Comment:
I'm not really a fan of how you're getting all of the meeting cards in this test, we should be able to use [`react-test-renderer`](https://reactjs.org/docs/test-renderer.html) to find components by their react types instead of selecting by text, which should be able to fix this and provide us with some useful queries when writing tests going forward, but let me know what you think

Reply:
What's the difference between react-test-renderer and Enzyme?

Reply:
In the latest commit, I attempt to resolve this issue within the Testing Library framework by using Test ID. Let me know if this is acceptable, or if you would like to move forward with `react-test-renderer`

Reply:
That's fine, the reason I didn't say to use test ids is that I remember you saying something along the lines of not wanting to use test ids since they'll only be used for tests, but I'm okay with using them if you are

Reply:
Tbh I don't like the look of test id's too much, but if we don't use them too often I think it'll be fine